include/tins/sniffer.h

```diff
@@ -45,6 +45,9 @@
 
 #include <pcap.h>
 
+using pcap_sniffing_method_t = int(*)(pcap_t*, int, pcap_handler, u_char*);
```

Comment:
That's what I did before changing my mind again: carrying around a value from an `enum` and making sure that it's either `pcap_loop` or `pcap_dispatch`. Why did I change my mind and used a plain function ptr? Two reasons:
1) passing just a function ptr is simpler. It is also riskier, and requires to import `pcap.h`, but it is only necessary if you want to use something that is not the default `pcap_loop`
2) it is more flexible: I can pass a custom function that can wrap around `pcap_loop` or `pcap_dispatch`. Could be for unit testing, for implementing a different sniffing backend or whatever other reason, it's just more flexible. I personally like this approach and I have some idea for the future that would use this approach.

What do you think?